.github/actions/e2e/cleanup/action.yaml

```diff
@@ -30,6 +30,18 @@ runs:
     - uses: ./.github/actions/e2e/install-eksctl
       with:
         eksctl_version: v0.147.0
+    - name: delete-instance-profiles
+      shell:
+      run: |
+        for name in $(aws iam list-instance-profiles --query "InstanceProfiles[*].{Name:InstanceProfileName}" --output text); do
+          tags=$(aws iam list-instance-profile-tags --instance-profile-name $name --output json)
+          if [[ $(echo $tags | jq -r '.Tags[] | select(.Key == "karpenter.sh/discovery") | .Value') == "${{ inputs.cluster_name }}" ]]; then 
+              roleName=$(aws iam get-instance-profile --instance-profile-name $name --query "InstanceProfile.Roles[*].{Name:RoleName}" --output text)
```

Comment:
Do we need to get the `RoleName`? Aren't we aware of the `RoleName` since we generate it in our testing?

Reply:
I was thinking that this could be more general by looking up the `RoleName`. If we need to create more instance profiles for testing, we would just need to tag the profile with `testing.karpenter.sh/discovery`